semgrep/semgrep/config_resolver.py

```diff
@@ -182,14 +183,14 @@ def download_config(config_url: str) -> Dict[str, Optional[Dict[str, Any]]]:
             if content_type and "text/plain" in content_type:
                 return parse_config_string("remote-url", r.content.decode("utf-8"))
             elif content_type and content_type == "application/x-gzip":
-                fname = f"/tmp/{base64.b64encode(config_url.encode()).decode()}"
-                shutil.rmtree(fname, ignore_errors=True)
-                with tarfile.open(fileobj=r.raw, mode="r:gz") as tar:
-                    tar.extractall(fname)
-                extracted = Path(fname)
-                for path in extracted.iterdir():
-                    # get first folder in extracted folder (this is how GH does it)
-                    return parse_config_folder(path, relative=True)
+                directory_name = base64.b64encode(config_url.encode()).decode()
+                with tempfile.TemporaryDirectory(suffix=directory_name) as fname:
```

Comment:
Is this suffix useful? Maybe we don't need to keep the b64 line, cause the directory name will still have a random component anyway:

```python
In [3]: tempfile.TemporaryDirectory(suffix="arst").name
Out[3]: '/var/folders/vp/mgl7xf9142x8wd081g9zrh2c0000gn/T/tmpdu1cx064arst'

In [4]: tempfile.TemporaryDirectory(prefix="arst").name
Out[4]: '/var/folders/vp/mgl7xf9142x8wd081g9zrh2c0000gn/T/arstwg4yijot'

In [5]: tempfile.TemporaryDirectory(prefix="arst", suffix="qwfp").name
Out[5]: '/var/folders/vp/mgl7xf9142x8wd081g9zrh2c0000gn/T/arstna4_ex81qwfp'
```

Reply:
Good call @underyx , addressed that